packages/client/src/provider/in-memory-provider/in-memory-provider.ts

```diff
@@ -0,0 +1,186 @@
+import {
+  EvaluationContext,
+  FlagNotFoundError,
+  FlagValueType,
+  GeneralError,
+  JsonValue,
+  Logger,
+  OpenFeatureError,
+  OpenFeatureEventEmitter,
+  ProviderEvents,
+  ResolutionDetails,
+  StandardResolutionReasons,
+  TypeMismatchError,
+  ProviderStatus,
+  Flag,
+  FlagConfiguration,
+  VariantFoundError,
+} from '@openfeature/shared';
+import { Provider } from '../provider';
+
+/**
+ * A simple OpenFeature provider intended for demos and as a test stub.
+ */
+export class InMemoryProvider implements Provider {
+  public readonly events = new OpenFeatureEventEmitter();
+  public readonly runsOn = 'client';
+  status: ProviderStatus = ProviderStatus.NOT_READY;
+  readonly metadata = {
+    name: 'in-memory',
+  } as const;
+  private _flagConfiguration: FlagConfiguration;
+  private _flagEvalCache: Record<string, ResolutionDetails<number | string | boolean | JsonValue>> = {};
+
+  constructor(flagConfiguration: FlagConfiguration = {}) {
+    this._flagConfiguration = { ...flagConfiguration };
+  }
+
+  async initialize(context?: EvaluationContext | undefined): Promise<void> {
+    this.status = ProviderStatus.READY;
+    try {
+
+      for (const key in this._flagConfiguration) {
+        if (this._flagConfiguration[key].disabled) {
+          continue;
+        }
+
+        this._flagEvalCache[key] = this.resolveFlagWithReason(key, context);
+
+      }
+
+      this.events.emit(ProviderEvents.Ready);
+    } catch (error) {
+      this.status = ProviderStatus.ERROR;
+      this.events.emit(ProviderEvents.Error);
+      throw error;
+    }
+  }
+
+  /**
+   * Overwrites the configured flags.
+   * @param { FlagConfiguration } flagConfiguration new flag configuration
+   */
+  putConfiguration(flagConfiguration: FlagConfiguration) {
+    const flagsChanged = Object.entries(flagConfiguration)
+      .filter(([key, value]) => this._flagConfiguration[key] !== value)
+      .map(([key]) => key);
+
+    this.status = ProviderStatus.STALE;
+    this.events.emit(ProviderEvents.Stale);
+
+    this._flagConfiguration = { ...flagConfiguration };
+    this.events.emit(ProviderEvents.ConfigurationChanged, { flagsChanged });
+  }
+
+  /**
+   * Overwrites the configured flags & updates internal cache (resolved flags).
+   * @param { FlagConfiguration } flagConfiguration new flag configuration
+   * @param { EvaluationContext } [context] flag evaluation context
+   */
+  async putConfigurationAndUpdateResolution(flagConfiguration: FlagConfiguration, context?: EvaluationContext | undefined): Promise<void> {
```

Comment:
When should a user use this method instead of `putConfiguration`?

Reply:
I think this question is also related to the previous one.

My idea was to:
1. Provide a method in which people could force provider's state to be stale (probably valuable while testing)
2. Provide another straight forward method to change flag configuration

Reply:
I also find this a little confusing. 
To me it feels that besides from changing the behavior in [line 130](https://github.com/open-feature/js-sdk/pull/617/files#diff-fa6dedaedba26b152e2aa7fa49d5e27aa197def6d2334aef668dba9b8a5d4a0aR130) if some flag is disabled,`putConfiguration` does not change the flag resolution, which means that I would alwys only use `putConfigurationAndUpdateResolution`. 
My feeling is that we only need one method that always changes the resolved flags as this method does.

Reply:
> Provide a method in which people could force provider's state to be stale (probably valuable while testing)

This actually behaves a bit odd here. As stated in the other comment, the only thing that is change here is the disabled state. The rest of the definitions does not change for the user. Maybe we could just have a method for setting the provider to stale without changing the data, if we want to support the use case  you described.

Reply:
> Provide another straight forward method to change flag configuration

The problem is, that `putConfiguration` does not change the flag configuration observed by the user (besides disabled status) if I see correctly.

Reply:
That's right! It does not change the cache, so the evaluation `value` is the same.
But since it sets `status=STALE` the reason will be different